ecs/pod.go

```diff
@@ -144,27 +145,29 @@ func (p *BasicECSPod) Stop(ctx context.Context) error {
 
 // Delete deletes the pod and its owned resources.
 func (p *BasicECSPod) Delete(ctx context.Context) error {
-	if err := p.Stop(ctx); err != nil {
-		return errors.Wrap(err, "stopping pod")
-	}
+	catcher := grip.NewBasicCatcher()
+
+	catcher.Wrap(p.Stop(ctx), "stopping pod")
 
 	if utility.FromBoolPtr(p.resources.TaskDefinition.Owned) {
-		deregisterDef := ecs.DeregisterTaskDefinitionInput{}
+		var deregisterDef ecs.DeregisterTaskDefinitionInput
 		deregisterDef.SetTaskDefinition(utility.FromStringPtr(p.resources.TaskDefinition.ID))
 
 		if _, err := p.client.DeregisterTaskDefinition(ctx, &deregisterDef); err != nil {
-			return errors.Wrap(err, "deregistering task definition")
+			catcher.Wrap(err, "deregistering task definition")
 		}
 	}
 
 	for _, secret := range p.resources.Secrets {
 		if utility.FromBoolPtr(secret.Owned) {
-			if err := p.vault.DeleteSecret(ctx, utility.FromStringPtr(secret.Name)); err != nil {
-				return errors.Wrap(err, "deleting secret")
-			}
+			catcher.Wrap(p.vault.DeleteSecret(ctx, utility.FromStringPtr(secret.Name)), "deleting secret")
 		}
 	}
 
+	if catcher.HasErrors() {
+		return catcher.Resolve()
+	}
+
 	p.status = cocoa.StatusDeleted
```

Comment:
I tested them empirically, and all the operations are idempotent. Since resources don't seem to get deleted (i.e. deregistered task definitions and stopped tasks are inactive but the resource metadata is still there), I think it's fine to assume that each AWS call is safe to retry.

Reply:
In that case, couldn't the same effect be accomplished by just removing [the check for running](https://github.com/evergreen-ci/cocoa/blob/ef4d1708f4ee50c635a6159f1bc4a0a35b57d425/ecs/pod.go#L129-L131)?

Reply:
I don't understand what you mean. I did remove that check, which was preventing stop from being idempotent. The entire function is idempotent, meaning the only way to reach the deleted state is if everything is truly cleaned up without error. It's safe to retry `Delete` as many times as you want until it returns no error, at which point you can be confident that everything's cleaned up.